01-builder-visualization/exercises/twitter-dogs.md

```diff
@@ -0,0 +1,199 @@
+# Twitter Pop-Ups & Time Slider Visualization
+
+* *Degree of Difficulty*: :star::star::star:
+* *Goal*: Create a time-enabled visualization that includes useful pop-ups using data pulled from Twitter with BUILDER
+* *Features Highlighted*:
+  * Style:
+    - Enabling time-based elements of your data for advanced visualization;
+    - Making the most of your data by adding and altering pop-ups to display useful information at a glance
+  * Layers: how to work with layer nodes
+  * Widgets: Formulas + Histograms
+* *Datasests needed*:
+  * `twitter_dogs` dataset: twitter data on based on dog breed search terms from 6/11/2017 - 7/10/17.
+
+
+1. [Import and create map](#map) <br>
+2. [Style](#sty) <br>
+3. [Widgets](#widgets) <br>
+4. [Publish and share map](#pub) <br>
+
+<hr>
+
+## 1. Import and create map <a name="map"></a>
+
+### 1. 1. Import `twitter_dogs` file into your DATASETS dashboard.
+
+* Download the dataset from [here](https://builder-demo.carto.com/api/v2/sql?q=SELECT+*+FROM+twitter_dogs&format=csv&filename=twitter_dogs)
+* Go to your account and import it using the **NEW DATASET** options
+* Select **CONNECT DATASET**
+* Drag and drop the file
+
+### 1. 2. Dataset view
+
+* Take a look on the dataset
+* Take note of which columns seem to be the most useful/interesting things in the dataset
+
+### 1. 3. Click on **CREATE MAP**
+
+* Change the name of the map to **Dogs of Twitter** (double click!)
+* Change the name of the layer to **Dogs of Twitter**
+* Change the Basemap to **Dark Matter (Labels Below)**
+
+<hr>
+
+## 2. Style <a name="sty"></a>
+
+### 2. 1. Styling Time-Based Elements
+
+* After clicking on **Dogs of Twitter** layer, it will direct you to the `STYLE` tab.
+* Change the **aggregation style** to `Animated`
+* Set the style options as follows:
+  * **FILL**:click on the color, click on `BY VALUE`, style by `category_terms`. Color these as you see fit by clicking on the color in line with the term you want to change.
+  * **COLUMN**: `postedtime`
+  * **DURATION**: `30`
+  * **STEPS**: `256`
+  * **TRAILS**: `2`
+  * **RESOLUTION**: `4`
+
+![animated settings](imgs/01-dog-01.png)
+
+* Click on the `LEGEND` tab. Set the style to `CUSTOM`. Check the box to add title and enter `By breed` to give your legend enough detail to be useful.
+
+### 2. 2. Styling Pop-ups
+
+* In order to add pop-ups to our map, we will need to add the `twitter_dogs` layer again. In the top left corner, press the back arrow to navigate back to the `LAYERS` menu.
```

Comment:
I would add some additional context here as to why we have to add the popups to a second layer, something like:

Pop ups can't be added to animated map layer, so will need to add our Twitter layer again as a static data layer to enable pop ups on that layer.